rust/capture/src/config.rs

```diff
@@ -61,6 +61,55 @@ pub struct Config {
     pub capture_mode: CaptureMode,
 
     pub concurrency_limit: Option<usize>,
+
+    #[envconfig(default = "false")]
+    pub s3_fallback_enabled: bool,
+    pub s3_fallback_bucket: Option<String>,
+    pub s3_fallback_endpoint: Option<String>,
+}
+
+impl Config {
+    pub fn from_env() -> Self {
+        Self {
+            print_sink: false,
+            address: "127.0.0.1:3000".parse().unwrap(),
+            redis_url: String::new(),
+            otel_url: None,
+            overflow_enabled: false,
+            overflow_per_second_limit: NonZeroU32::new(100).unwrap(),
+            overflow_burst_limit: NonZeroU32::new(1000).unwrap(),
+            overflow_forced_keys: None,
+            dropped_keys: None,
+            kafka: KafkaConfig {
```

Comment:
Isn't this a duplicate of the stuff below 🤔

Reply:
Yeah this needs a refactor tbh